### ISSUE TITLE
Chromium building errors

### ISSUE THREAD

**Issue by faulesocke** [voidlinux/void-packages#7599]
Hey there,

I just tried building chromium from sources:
`./xbps-src pkg -j6 chromium`

and got the following errors during the build phase:
```
FAILED: obj/breakpad/client/microdump_writer.o 
g++ -MMD -MF obj/breakpad/client/microdump_writer.o.d -DV8_DEPRECATION_WARNINGS -DUSE_UDEV -DUI_COMPOSITOR_IMAGE_TRANSPORT -DUSE_AURA=1 -DUSE_PANGO=1 -DUSE_CAIRO=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -DUSE_X11=1 -DNO_TCMALLOC -DDISABLE_NACL -DFULL_SAFE_BROWSING -DSAFE_BROWSING_CSD -DSAFE_BROWSING_DB_LOCAL -DCHROMIUM_BUILD -DENABLE_MEDIA_ROUTER=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -I../../breakpad -I../../breakpad/src -I../../breakpad/src/client -I../../breakpad/src/third_party/linux/include -I../.. -Igen -I../../breakpad/src -fno-strict-aliasing --param=ssp-buffer-size=4 -fstack-protector -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -funwind-tables -fPIC -pipe -B/usr/bin -m64 -march=x86-64 -pthread -O2 -fno-ident -fdata-sections -ffunction-sections -g0 -fvisibility=hidden -Wno-unused-local-typedefs -Wno-maybe-uninitialized -Wno-missing-field-initializers -Wno-unused-parameter -fvisibility-inlines-hidden -std=gnu++11 -fno-delete-null-pointer-checks -fno-rtti -fno-exceptions -Wno-deprecated -Wno-narrowing -c ../../breakpad/src/client/linux/microdump_writer/microdump_writer.cc -o obj/breakpad/client/microdump_writer.o
In file included from ../../breakpad/src/client/linux/microdump_writer/microdump_writer.cc:41:0:
../../breakpad/src/client/linux/handler/exception_handler.h:194:21: error: field 'context' has incomplete type 'google_breakpad::ucontext'
     struct ucontext context;
                     ^~~~~~~
In file included from ../../breakpad/src/client/linux/microdump_writer/microdump_writer.cc:40:0:
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:44:49: note: forward declaration of 'struct google_breakpad::ucontext'
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                                                 ^~~~~~~~
../../breakpad/src/client/linux/microdump_writer/microdump_writer.cc: In member function '{anonymous}::MicrodumpWriter::CaptureResult {anonymous}::MicrodumpWriter::CaptureCrashingThreadStack(int)':
../../breakpad/src/client/linux/microdump_writer/microdump_writer.cc:245:63: error: no matching function for call to 'google_breakpad::UContextReader::GetStackPointer(const {anonymous}::ucontext* const&)'
     stack_pointer_ = UContextReader::GetStackPointer(ucontext_);
                                                               ^
In file included from ../../breakpad/src/client/linux/microdump_writer/microdump_writer.cc:40:0:
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:44:20: note: candidate: static uintptr_t google_breakpad::UContextReader::GetStackPointer(const google_breakpad::ucontext*)
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                    ^~~~~~~~~~~~~~~
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:44:20: note:   no known conversion for argument 1 from 'const {anonymous}::ucontext* const' to 'const google_breakpad::ucontext*'
../../breakpad/src/client/linux/microdump_writer/microdump_writer.cc:270:67: error: no matching function for call to 'google_breakpad::UContextReader::GetInstructionPointer(const {anonymous}::ucontext* const&)'
     uintptr_t pc = UContextReader::GetInstructionPointer(ucontext_);
                                                                   ^
In file included from ../../breakpad/src/client/linux/microdump_writer/microdump_writer.cc:40:0:
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:46:20: note: candidate: static uintptr_t google_breakpad::UContextReader::GetInstructionPointer(const google_breakpad::ucontext*)
   static uintptr_t GetInstructionPointer(const struct ucontext* uc);
                    ^~~~~~~~~~~~~~~~~~~~~
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:46:20: note:   no known conversion for argument 1 from 'const {anonymous}::ucontext* const' to 'const google_breakpad::ucontext*'
../../breakpad/src/client/linux/microdump_writer/microdump_writer.cc: In member function 'void {anonymous}::MicrodumpWriter::DumpCPUState()':
../../breakpad/src/client/linux/microdump_writer/microdump_writer.cc:402:65: error: no matching function for call to 'google_breakpad::UContextReader::FillCPUContext(google_breakpad::RawContextCPU*, const {anonymous}::ucontext* const&, const fpstate_t* const&)'
     UContextReader::FillCPUContext(&cpu, ucontext_, float_state_);
                                                                 ^
In file included from ../../breakpad/src/client/linux/microdump_writer/microdump_writer.cc:40:0:
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:52:15: note: candidate: static void google_breakpad::UContextReader::FillCPUContext(google_breakpad::RawContextCPU*, const google_breakpad::ucontext*, const _libc_fpstate*)
   static void FillCPUContext(RawContextCPU *out, const ucontext *uc,
               ^~~~~~~~~~~~~~
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:52:15: note:   no known conversion for argument 2 from 'const {anonymous}::ucontext* const' to 'const google_breakpad::ucontext*'
[308/26780] CXX obj/base/base/lock_impl_posix.o
[309/26780] CXX obj/breakpad/client/minidump_writer.o
FAILED: obj/breakpad/client/minidump_writer.o 
g++ -MMD -MF obj/breakpad/client/minidump_writer.o.d -DV8_DEPRECATION_WARNINGS -DUSE_UDEV -DUI_COMPOSITOR_IMAGE_TRANSPORT -DUSE_AURA=1 -DUSE_PANGO=1 -DUSE_CAIRO=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -DUSE_X11=1 -DNO_TCMALLOC -DDISABLE_NACL -DFULL_SAFE_BROWSING -DSAFE_BROWSING_CSD -DSAFE_BROWSING_DB_LOCAL -DCHROMIUM_BUILD -DENABLE_MEDIA_ROUTER=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -I../../breakpad -I../../breakpad/src -I../../breakpad/src/client -I../../breakpad/src/third_party/linux/include -I../.. -Igen -I../../breakpad/src -fno-strict-aliasing --param=ssp-buffer-size=4 -fstack-protector -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -funwind-tables -fPIC -pipe -B/usr/bin -m64 -march=x86-64 -pthread -O2 -fno-ident -fdata-sections -ffunction-sections -g0 -fvisibility=hidden -Wno-unused-local-typedefs -Wno-maybe-uninitialized -Wno-missing-field-initializers -Wno-unused-parameter -fvisibility-inlines-hidden -std=gnu++11 -fno-delete-null-pointer-checks -fno-rtti -fno-exceptions -Wno-deprecated -Wno-narrowing -c ../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc -o obj/breakpad/client/minidump_writer.o
In file included from ../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc:69:0:
../../breakpad/src/client/linux/handler/exception_handler.h:194:21: error: field 'context' has incomplete type 'google_breakpad::ucontext'
     struct ucontext context;
                     ^~~~~~~
In file included from ../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc:68:0:
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:44:49: note: forward declaration of 'struct google_breakpad::ucontext'
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                                                 ^~~~~~~~
../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc: In member function 'bool {anonymous}::MinidumpWriter::CrashingThreadReferencesPrincipalMapping()':
../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc:196:78: error: no matching function for call to 'google_breakpad::UContextReader::GetStackPointer(const {anonymous}::ucontext* const&)'
     const uintptr_t stack_pointer = UContextReader::GetStackPointer(ucontext_);
                                                                              ^
In file included from ../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc:68:0:
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:44:20: note: candidate: static uintptr_t google_breakpad::UContextReader::GetStackPointer(const google_breakpad::ucontext*)
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                    ^~~~~~~~~~~~~~~
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:44:20: note:   no known conversion for argument 1 from 'const {anonymous}::ucontext* const' to 'const google_breakpad::ucontext*'
../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc:197:73: error: no matching function for call to 'google_breakpad::UContextReader::GetInstructionPointer(const {anonymous}::ucontext* const&)'
     const uintptr_t pc = UContextReader::GetInstructionPointer(ucontext_);
                                                                         ^
In file included from ../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc:68:0:
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:46:20: note: candidate: static uintptr_t google_breakpad::UContextReader::GetInstructionPointer(const google_breakpad::ucontext*)
   static uintptr_t GetInstructionPointer(const struct ucontext* uc);
                    ^~~~~~~~~~~~~~~~~~~~~
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:46:20: note:   no known conversion for argument 1 from 'const {anonymous}::ucontext* const' to 'const google_breakpad::ucontext*'
../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc: In member function 'bool {anonymous}::MinidumpWriter::WriteThreadListStream(MDRawDirectory*)':
../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc:418:78: error: no matching function for call to 'google_breakpad::UContextReader::GetStackPointer(const {anonymous}::ucontext* const&)'
         const uintptr_t stack_ptr = UContextReader::GetStackPointer(ucontext_);
                                                                              ^
In file included from ../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc:68:0:
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:44:20: note: candidate: static uintptr_t google_breakpad::UContextReader::GetStackPointer(const google_breakpad::ucontext*)
   static uintptr_t GetStackPointer(const struct ucontext* uc);
                    ^~~~~~~~~~~~~~~
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:44:20: note:   no known conversion for argument 1 from 'const {anonymous}::ucontext* const' to 'const google_breakpad::ucontext*'
../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc:420:77: error: no matching function for call to 'google_breakpad::UContextReader::GetInstructionPointer(const {anonymous}::ucontext* const&)'
                              UContextReader::GetInstructionPointer(ucontext_),
                                                                             ^
In file included from ../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc:68:0:
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:46:20: note: candidate: static uintptr_t google_breakpad::UContextReader::GetInstructionPointer(const google_breakpad::ucontext*)
   static uintptr_t GetInstructionPointer(const struct ucontext* uc);
                    ^~~~~~~~~~~~~~~~~~~~~
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:46:20: note:   no known conversion for argument 1 from 'const {anonymous}::ucontext* const' to 'const google_breakpad::ucontext*'
../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc:426:70: error: no matching function for call to 'google_breakpad::UContextReader::GetInstructionPointer(const {anonymous}::ucontext* const&)'
         uint64_t ip = UContextReader::GetInstructionPointer(ucontext_);
                                                                      ^
In file included from ../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc:68:0:
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:46:20: note: candidate: static uintptr_t google_breakpad::UContextReader::GetInstructionPointer(const google_breakpad::ucontext*)
   static uintptr_t GetInstructionPointer(const struct ucontext* uc);
                    ^~~~~~~~~~~~~~~~~~~~~
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:46:20: note:   no known conversion for argument 1 from 'const {anonymous}::ucontext* const' to 'const google_breakpad::ucontext*'
../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc:472:74: error: no matching function for call to 'google_breakpad::UContextReader::FillCPUContext(MDRawContextAMD64*, const {anonymous}::ucontext* const&, const fpstate_t* const&)'
         UContextReader::FillCPUContext(cpu.get(), ucontext_, float_state_);
                                                                          ^
In file included from ../../breakpad/src/client/linux/minidump_writer/minidump_writer.cc:68:0:
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:52:15: note: candidate: static void google_breakpad::UContextReader::FillCPUContext(google_breakpad::RawContextCPU*, const google_breakpad::ucontext*, const _libc_fpstate*)
   static void FillCPUContext(RawContextCPU *out, const ucontext *uc,
               ^~~~~~~~~~~~~~
../../breakpad/src/client/linux/dump_writer_common/ucontext_reader.h:52:15: note:   no known conversion for argument 2 from 'const {anonymous}::ucontext* const' to 'const google_breakpad::ucontext*'
[310/26780] CXX obj/breakpad/client/linux_core_dumper.o
[311/26780] CXX obj/breakpad/client/linux_ptrace_dumper.o
[312/26780] CXX obj/breakpad/client/linux_dumper.o
ninja: build stopped: subcommand failed.
=> ERROR: chromium-58.0.3029.110_1: failed to run do_build() at line 220.
```

I updated my system and re-cloned the void-packages repo to a new location but the error still occours.

Any ideas?

**Comment by ebfe** [comment]
This is due to the recent glibc 2.26 changes, there are patches to fix these in #7164

**Comment by faulesocke** [comment]
Thanks for the quick answer, looks like the patch mentioned there works after some little modifications. I've create pull request #7605 which yields a working chromium.
